qiskit/primitives/containers/bindings_array.py

```diff
@@ -33,66 +33,50 @@
-    format: ``[vals, vals1[..., :10], vals1[..., 10:], ...]`` is exactly equivalent to
-    ``[vals0, vals1, ...]`` in the bindings it specifies. This allows flexibility about whether
-    values for different parameters are stored in one big array, or across several smaller
-    arrays. It also allows different parameters to use different dtypes.
-
-    .. code-block:: python
-
-        # 0-d array (i.e. only one binding)
-        BindingsArray([1, 2, 3], {"a": 4, ("b", "c"): [5, 6]})
-
-        # single array, last index is parameters
-        BindingsArray(np.empty((10, 10, 100)))
-
-        # multiple arrays, where each last index is parameters. notice that it's smart enough to
-        # figure out that a missing last dimension corresponds to a single parameter.
-        BindingsArray(
-            [np.empty((10, 10, 100)), np.empty((10, 10)), np.empty((10, 10, 20), dtype=complex)],
-            {("c", "a"): np.empty((10, 10, 2)), "b": np.empty((10, 10))}
-        )
+        A single parameter binding set provides numeric values to bind to a circuit with free
+        :class:`qiskit.circuit.Parameter`\s. An instance of this class stores an array-valued collection
+        of such sets. The simplest example is a 0-d array consisting of a single parameter binding set,
+        whereas an n-d array of parameter binding sets represents an n-d sweep over values.
+
+        The storage format is a dictionary of arrays attached to parameters, ``{params_0: values_0,
+        ...}``. A convention is used where the last dimension of each array indexes (a subset of)
+        circuit parameters. For example, if the last dimension of ``values_0`` is 25, then it represents
+        an array of possible binding values for the 25 distinct parameters ``params_0``, where its
+        leading shape is the array :attr:`~.shape` of its binding array. This allows flexibility about
+        whether values for different parameters are stored in one big array, or across several smaller
+        arrays. It also allows different parameters to use different dtypes.
+
+        .. code-block:: python
+
+            # 0-d array (i.e. only one binding)
+            BindingsArray({"a": 4, ("b", "c"): [5, 6]})
+    ß
+            # single array, last index is parameters
+            parameters = tuple(f"a{idx}" for idx in range(100))
+            BindingsArray({parameters: np.empty((10, 10, 100))})
+
+            # multiple arrays, where each last index is parameters. notice that it's smart enough to
+            # figure out that a missing last dimension corresponds to a single parameter.
+            BindingsArray(
+                {("c", "a"): np.empty((10, 10, 2)), "b": np.empty((10, 10), dtype=int)}
+            )
     """
-    __slots__ = ("_vals", "_kwvals")
 
     def __init__(
         self,
-        vals: ArrayLike | Iterable[ArrayLike] | None = None,
         kwvals: Mapping[ParameterLike, Iterable[ParameterValueType]] | ArrayLike | None = None,
```

Comment:
Two questions:

1. Do we want to keep this arg called `kwvals` now that there is no `vals` arg
2. Do we want to keep it with a default value of None so you can do `BindingsArray()` for zero parameter arrays. Or do we make it a positional arg so you would have to do `BindingsArray(None)` or `BindingsArray({})` to make a zero parameter array?

I dont have a strong preference either way for these

Reply:
My preference is to support `BindingsArray()` to have a 0d array of 0 parameters, but like you, this is not something I will fight about. So since this is the way it already is in this PR, I'd say to just leave it as-is.

Reply:
(the commit didn't have all of the changes it needed, hold on)

Reply:
Okay, pushed. I'm not sure what happened. The tests I fixed yesterday vanished and I had to do it again.